wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -989,6 +989,25 @@ public function delete() {
 		$this->disableUpdate();
 	}
 	
+	/**
+	 * Deletes this session if:
+	 * - it is newly created in this request, and
+	 * - it belongs to a guest.
+	 * 
+	 * This method is useful if you have controllers that are likely to be
+	 * accessed by a user agent that is not going to re-use sessions (e.g.
+	 * curl in a cronjob). It immediately remove the session that was created
+	 * just for that request and that is not going to be used ever again.
+	 * 
+	 * @since 5.2
+	 */
+	public function deleteIfNew() {
```

Comment:
This is a very short function and the use of two ifs and an implicit else make it much harder to read. I would like to see a single if instead that evaluates to true and then deletes the session.